experiments/krico/workloads/caching.go

```
@@ -0,0 +1,218 @@
+	maxQPS := sensitivity.PeakLoadFlag.Value()
+
+	//	Prepare executor.
+	loadGeneratorExecutorConfig := executor.DefaultRemoteConfig()
+	loadGeneratorExecutor, err := executor.NewRemote(aggressorAddress.Value(), loadGeneratorExecutorConfig)
+	errutil.CheckWithContext(err, "Cannot prepare YCSB Redis executor !")
+
+	//	Prepare launcher.
+	ycsb.CalculateWorkloadCommandParameters(maxQPS, loadDuration, &loadGeneratorConfig)
+	loadGeneratorLauncher := ycsb.New(loadGeneratorExecutor, loadGeneratorConfig)
+
+	//	Populate workload.
+	err = loadGeneratorLauncher.Populate()
+	errutil.CheckWithContext(err, "Cannot load the initial test data into Redis!")
+
+	//
+	//	Metrics collector (Snap)
+	//
+
+	//	Start Snap on OpenStack host.
+	StartSnapService(workloadExecutorConfig.Hypervisor.Address)
+
+	//	Obtain workload instance cgroup.
+	cgroup, err := GetInstanceCgroup(workloadExecutorConfig.Hypervisor.InstanceName, workloadExecutorConfig.Hypervisor.Address)
+	errutil.CheckWithContext(err, "Cannot obtain workload instance cgroup !")
+
+	snapTaskConfig := kricosnapsession.DefaultConfig(cgroup, workloadExecutorConfig.Hypervisor.InstanceName)
+	snapTaskConfig.Tags = PrepareDefaultKricoTags(workloadExecutorConfig, experimentID)
+
+	//	Prepare launcher.
+	snapTaskLauncher, err := kricosnapsession.NewSessionLauncher(snapTaskConfig)
+	errutil.CheckWithContext(err, "Cannot obtain Snap Task Launcher !")
+
+	//	Run Snap task.
+	snapTaskHandle, err := snapTaskLauncher.Launch()
+	errutil.CheckWithContext(err, "Cannot gather performance metrics !")
+
+	//	Stop snap task on the end of experiment.
+	defer snapTaskHandle.Stop()
+
+	//
+	//	Load Generator
+	//
+
+	//	Start load on workload.
+	loadGeneratorHandle, err := loadGeneratorLauncher.Load(maxQPS, loadDuration)
+	errutil.CheckWithContext(err, "Cannot start YCSB Redis task !")
+
+	//	In the end stop load on workload.
+	defer loadGeneratorHandle.Stop()
+
+	//	Wait until load generating finishes.
+	loadGeneratorHandle.Wait(0)
```

Do not ignore errors, please.

---

Error is still being ignored.